caching-big-repositories_5bf03fb92c7d3a31944e139d.md

```diff
@@ -0,0 +1,107 @@
+
+* [Overview](#overview)
+* [The logic behind the recipe](#the-logic-behind-the-recipe)
+* [An example project](#an-example-project)
+* [See Also](#see-also)
+
+## Overview
+
+This guide shows you how to cache a large Git repository and avoid cloning it
+from scratch in every block.
+
+Note that for this recipe to work, the files of the GitHub repository should
+remain intact during the execution of the pipeline or you will end up having
+a different version of the files in the GitHub repository and in the Semaphore
+Cache server.
+
+## The logic behind the recipe
+
+The example project that follows is getting all the files of the GitHub
+repository in the first job using `checkout` and then is storing them into the
+Semaphore 2.0 Cache server using `cache store`. The remaining jobs of the
+project will use these files from the Cache server using `cache restore`
+without calling `checkout` again.
+
+## An example project
+
+The example project of this section will use 10 large files. Each one of these
+10 files is around 100MB in size.
+
+Each one file was generated using the following command:
+
+    dd if=/dev/random of=filename bs=1024 count=100000
+
+The Semaphore 2.0 environment variables that will be used are the following:
+
+* `SEMAPHORE_PROJECT_NAME`: This environment variable holds the name of the
+    current Semaphore 2.0 project.
+* `SEMAPHORE_GIT_SHA`: This environment variable holds the current revision of
+    the GitHub code that the pipeline is using.
+
+Note that the combination of `SEMAPHORE_PROJECT_NAME` and `SEMAPHORE_GIT_SHA`
+gets updated each time you make changes to the GitHub repository.
+
+The `semaphore.yml` file for the example project will be the following:
+
+	version: v1.0
+	name: Recipe of Caching Big Repositories
+	agent:
+	  machine:
+	    type: e1-standard-2
+	    os_image: ubuntu1804
+    
+	blocks:
+	  - name: Reuse GitHub repository
+	    task:
+	      jobs:
+	      - name: Cache GitHub repository
+	        commands:
+	          - checkout
+	          - cd ..
+	          - cache store "$SEMAPHORE_PROJECT_NAME-$SEMAPHORE_GIT_SHA" $SEMAPHORE_PROJECT_NAME
+    
+	  - name: Task 2 - Single Job
+	    task:
+	      jobs:
+	      - name: Use Cache - Job 1
+	        commands:
+	          - cache restore "$SEMAPHORE_PROJECT_NAME-$SEMAPHORE_GIT_SHA"
+	          - ls -al "$SEMAPHORE_PROJECT_NAME"/FILES
+    
+	  - name: Task 3 - Single Job
```

Comment:
would reader lose some understanding if we removed blocks 3, 4 and 5?

Reply:
And add at least one more job to Task 2.
Otherwise the example is meaningless.

As a side-note: entities named "Task #" are Blocks and should be named that to avoid confusion.

Reply:
Renamed `Task #` to `Block item` and added another Block item at the end.